parking/backend/sensor_server/rest_server.py

```diff
@@ -0,0 +1,81 @@
+import json
+from tornado import web
+from parking.shared.rest_models import (ParkingLot, ParkingLotCreationResponse, ParkingLotAvailableMessage,
+                                        ParkingLotPriceMessage, InvalidLotFieldsError, InvalidLotIDError,
+                                        InvalidLotAvailableError, RESTError, InvalidLotPriceError)
+from parking.shared.util import serialize_model
+from parking.backend.db.dbaccess import DbAccess
+
+
+class ParkingLotsHandler(web.RequestHandler):
+    def initialize(self, dba: DbAccess) -> None:
+        self.dba = dba
+
+    async def post(self):
+        try:
+            lot = ParkingLot(**json.loads(self.request.body))
+        except (TypeError, ValueError) as err:
+            self.write(serialize_model(self._create_error_response(err)))
+        else:
+            pid = await self.dba.insert_parking_lot(lot)
+            self.write(serialize_model(ParkingLotCreationResponse(id=pid)))
+
+    def _create_error_response(self, err: Exception) -> ParkingLotCreationResponse:
+        if isinstance(err, ValueError):
+            return ParkingLotCreationResponse(error=InvalidLotFieldsError(str(err)))
+        else:
+            return ParkingLotCreationResponse(error=InvalidLotFieldsError('Invalid or missing fields'))
+
+
+class IndividualLotHandler(web.RequestHandler):
+    def initialize(self, dba: DbAccess) -> None:
+        self.dba = dba
+
+    def prepare(self):
+        if self.request.headers.get("Content-Type", "").startswith("application/json"):
+            self.json_args = json.loads(self.request.body)
+        else:
+            self.json_args = None
+
+    def _create_error_response(self, err: Exception, parse_err_cls: RESTError) -> RESTError:
+        if isinstance(err, ValueError):
+            return parse_err_cls(str(err))
+        else:
+            return parse_err_cls('Invalid or missing fields')
+
+    async def post(self, lot_id: str):
+        if self.request.uri.endswith('available'):
+            await self.post_available(int(lot_id))
+        if self.request.uri.endswith('price'):
+            await self.post_price(int(lot_id))
+
+    def return_invalid_lot_id_err(self, lot_id: int):
+        self.clear()
+        self.set_status(400)
+        self.finish(serialize_model(InvalidLotIDError('Invalid lot ID: ' + str(lot_id))))
+
+    async def post_available(self, lot_id: int):
+        try:
+            msg = ParkingLotAvailableMessage(**self.json_args)
+        except (TypeError, ValueError) as err:
+            self.write(serialize_model(self._create_error_response(err, InvalidLotAvailableError)))
+        else:
+            park_id = await self.dba.update_parking_lot_availability(int(lot_id), msg.available)
```

Comment:
Similarly a wrapper around the DBAL to promote database errors into HTTPErrors would be valuable too

Reply:
Definitely, I've not handled DB errors yet